.github/workflows/main.yml

```diff
@@ -202,19 +209,21 @@ jobs:
           retention-days: 5
 
       - name: Download darwin release artifact
+        if: needs.mac-os-build.result == 'success'
         uses: actions/download-artifact@v3
         with:
           name: darwin-release-${{ github.run_id }}
           path: release/
 
       - name: Download windows release artifact
+        if: needs.windows-build.result == 'success'
         uses: actions/download-artifact@v3
         with:
           name: windows-release-${{ github.run_id }}
           path: release/
 
       - name: Install Jfrog CLI
-        uses: jfrog/setup-jfrog-cli@v2
+        uses: jfrog/setup-jfrog-cli@v3
```

Comment:
this one change seems minor but it worries me. sometimes things like this have unintended consequences.

Reply:
This is a response to GH's deprecation notice for Node.js v12. We should try v3 before they fully decommission. We used this change to the jfrog CLI setup action in the checks run on this PR, so we know that it works.